doc/src/Developer_write_pair.rst

```diff
@@ -0,0 +1,1196 @@
+then the extra communication is skipped, since for pairs straddling
+sub-domain boundaries, the forces are computed twice and only stored
+with the local atoms in the domain that *owns* it.
+
+.. code-block:: c++
+
+           f[i][0] += delx * fpair;
+           f[i][1] += dely * fpair;
+           f[i][2] += delz * fpair;
+           if (newton_pair || j < nlocal) {
+             f[j][0] -= delx * fpair;
+             f[j][1] -= dely * fpair;
+             f[j][2] -= delz * fpair;
+           }
+
+The ``ev_tally()`` function tallies global or per-atom energy and
+virial.  For typical MD simulations, the potential energy is merely a
+diagnostic and only needed on output.  Similarly, the pressure may only
+be computed for (infrequent) thermodynamic output.  For all timesteps
+where this information is not needed either, `eflag` or `evflag` are
+zero and the computation and call to the tally function skipped.  Note
+that evdwl is initialized to zero at the beginning of the function, so
+that it still is valid to access it, even if the energy is not computed
+(e.g. when only the virial is needed).
+
+.. code-block:: c++
+
+           if (eflag) evdwl = factor_lj * (aexp - bexp - offset[itype][jtype]);
+           if (evflag) ev_tally(i, j, nlocal, newton_pair, evdwl, 0.0, fpair, delx, dely, delz);
+         }
+       }
+     }
+
+If only the global virial is needed and no energy, then calls to
+``ev_tally()`` can be avoided altogether, and the global virial can be
+computed more efficiently from the dot product of the total per-atom
+force vector and the position vector of the corresponding atom,
+:math:`\vec{F}\cdot\vec{r}`.  This has to be done *after* all pair-wise
+forces are computed and *before* the reverse communication to collect
+data from ghost atoms, since the position has to be the position that was
+used to compute the force, i.e. *not* the "local" position if that ghost
+atom is a periodic copy.
+
+.. code-block:: c++
+
+     if (vflag_fdotr) virial_fdotr_compute();
+   }
+
+
+Computing force and energy for a single pair
+""""""""""""""""""""""""""""""""""""""""""""
+
+Certain features in LAMMPS only require computing interactions between
+individual pairs of atoms and the (optional) ``single()`` function is
+needed to support those features (e.g. for tabulation of force and
+energy with :doc:`pair_write <pair_write>`).  This is a repetition of
+the force kernel in the ``compute()`` function, but only for a single
+pair of atoms, where the (squared) distance is provided as a parameter
+(so it may not even be an existing distance between two specific atoms).
+The energy is returned as the return value of the function and the force
+as the `fforce` reference.  Note, that this is, similar to how *fpair*
+is used in the ``compute()`` function, the magnitude of the force along
+the vector between the two atoms *divided* by the distance.
+
+The ``single()`` function is optional.  The member variable
```

Comment:
Explain that any true pairwise style should define this method.  Manybody methods do not.